tools/xabuild/XABuildPaths.cs

```diff
@@ -141,16 +142,38 @@ public XABuildPaths ()
 				NuGetTargets             = Path.Combine (nuget, "Microsoft.NuGet.targets");
 				NuGetRestoreTargets      = Path.Combine (VsInstallRoot, "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet", "NuGet.targets");
 			} else {
-				const string vsVersion   = "15.0";
+				string[] vsVersions       = new [] {"Current", "15.0"};
 				string mono              = IsMacOS ? "/Library/Frameworks/Mono.framework/Versions/Current/lib/mono" : "/usr/lib/mono";
 				string monoExternal      = IsMacOS ? "/Library/Frameworks/Mono.framework/External/" : "/usr/lib/mono";
 				MSBuildPath              = Path.Combine (mono, "msbuild");
-				MSBuildBin               = Path.Combine (MSBuildPath, vsVersion, "bin");
+
+				MSBuildBin = null;
+				foreach (string vsVersion in vsVersions) {
+					MSBuildBin = Path.Combine (MSBuildPath, vsVersion, "bin");
+					if (Directory.Exists (MSBuildBin))
+						break;
+				}
+				if (String.IsNullOrEmpty (MSBuildBin))
```

Comment:
I think we normally use lowercase `string`? I have only seen the uppercase version in really old code.

Reply:
I always use the uppercase version...

Reply:
See: https://blog.paranoidcoding.com/2019/04/08/string-vs-String-is-not-about-style.html

Reply:
@jonathanpeppers it's not a matter of style for *me* but precisely of what I wrote - I think it's the correct way to call a static member of a class defined in the BCL, documented as such and more (IMO) correct than calling the same function on a *language* type. And the blog post is irrelevant to this particular use case. Nowhere I use, or would ever use, `String s;` instead of `string s;` as that would be abuse of the identity between `System.String` and `string` - and in this regard I agree with the blog post. But method calls on `System.String` which take a `string` as a parameter? I'll stick to my way of calling these methods :)